internal/singleflight/singleflight.go

```diff
@@ -0,0 +1,244 @@
+// Copyright 2013 The Go Authors. All rights reserved.
```

Comment:
Thinking out loud:
* Alternatively, move the license to LISENSE.md in root of the package? 
* If BSD is compatible with the go-f3 license I think we could skip this notice entirely maybe?

Not sure what the rules are.